crates/sui-core/src/authority_aggregator.rs

```diff
@@ -1053,29 +1059,51 @@ where
         }
     }
 
-    fn handle_transaction_response_with_signed(
+    fn handle_transaction_response_plain(
         &self,
         state: &mut ProcessTransactionState,
-        signed_transaction: VerifiedSignedTransaction,
+        plain_tx: SignedTransaction,
     ) -> SuiResult<Option<ProcessTransactionResult>> {
-        let (data, sig) = signed_transaction.into_inner().into_data_and_sig();
+        let (data, sig) = plain_tx.into_data_and_sig();
         match state.tx_signatures.insert(sig) {
             InsertResult::NotEnoughVotes => Ok(None),
             InsertResult::Failed { error } => Err(error),
             InsertResult::QuorumReached(cert_sig) => {
-                let ct = CertifiedTransaction::new_from_data_and_sig(data, cert_sig);
-                Ok(Some(ProcessTransactionResult::Certified(
-                    ct.verify(&self.committee)?,
-                )))
+                let ct = CertifiedTransaction::new_from_data_and_sig(data.clone(), cert_sig);
+                let res = ct.verify(&self.committee);
+                match res {
+                    Ok(ct) => Ok(Some(ProcessTransactionResult::Certified(ct))),
+                    Err(err) => {
+                        debug!(
+                            "Fail to verify the aggregated siganture for tx: {:?}",
+                            data.digest()
+                        );
+                        for (name, s) in &state.tx_signatures.clone().data {
```

Comment:
I'd add a comment that this is fallback, where we verify one-by-one and then we remove invalid signatures (and weights). Have a few other questions re removing faulty vs counting correct and omitting faulty. Will reach out.